tests/src/BuiltIn/AbstractBuiltInTest.php

```diff
@@ -0,0 +1,19 @@
+<?php declare(strict_types=1);
+
+namespace Room11\Jeeves\Tests\BuiltIn;
+
+use Room11\Jeeves\Chat\Client\ChatClient;
+
+abstract class AbstractBuiltInTest extends \PHPUnit\Framework\TestCase
+{
+    protected $client;
+
+    protected function setUp()
+    {
+        parent::setUp();
+
+        $this->client = $this->getMockBuilder(ChatClient::class)
+            ->disableOriginalConstructor()
+            ->getMock();
```

Comment:
@JayPHP maybe things like that could accept the `Room` as an arg rather than making it a property of the parent?

I had a slight wtf at some point when tests were referencing a prop from the parent, shared logic is fine but maybe not shared state? Converting those things to static stateless methods or bare functions seems like it makes for easier readability when it comes to tests.